software/grapherrific.livecodescript

```diff
@@ -0,0 +1,1477 @@
+   put tYfor & "," & tTurn into pMain[pDoc][tZback][pBackward]
+   put tTurn & "," & tTurn into pMain[pDoc][tZback]["myTurn"]
+   put pZstep & "," & tTurn into pMain[pDoc][tZback]["myStep"]
+   put cr & tZback & "," & tTurn after pMain[pDoc][pZstep]["step"]
+   
+   return tRec+2
+end newDilink_Old
+
+command changeInfo @pMain,pBlerg
+   put "command changeInfo" into xX["write"];    devLog xX
+   -- pBlerg["doc"] = doc in address
+   -- pBlerg["rec"] = step's record key
+   -- pBlerg["tag"] = tag string (this will be in a rec different from the step's rec)
+   -- pBlerg["data"] = new data
+   -- pBlerg["hide"] = true 
+   put pBlerg["rec"] into pRec
+   put pBlerg["tag"] into pTag
+   put pBlerg["doc"] into pDoc   
+   put pBlerg["data"] into pData
+   put pBlerg["hide"] into pHide
+   -- if the change is identical to existing data, don't do anything
+   -- if pHide is true then hide the [tag] in the [rec] in the [doc]
+   put pMain[pDoc][0][lastKey] into tRec -- this order makes it work better
+   if pHide is true then
+      -- negate the tag in the step [rec] that points to [tag]
+      -- negate the other end of the pointer
+      put getTagKey(pMain,pBlerg) into tOldTag
+      put cr & "-" & tOldTag & "," & tRec+1 after pMain[pDoc][pRec]["step"]
+      put cr & "-" & pRec & "," & tRec + 1 after pMain[pDoc][tOldTag]["myStep"]
+      -- create turn
+      put tRec+2 & "," & tRec+1 into pMain[pDoc][tRec+1]["turn"]
+      put "changeInfo" into pMain[pDoc][tRec+1]["act"]
+      put tOldTag & "," & tRec+1 into pMain[pDoc][tRec+2]["hide"]
+      put tRec+1 & "," & tRec+1 into pMain[pDoc][tRec+2]["myTurn"]
+   else
+      --      put pMain[pDoc][0][lastKey] into tRec -- this order makes it work better
+      put getTagData(pMain,pBlerg) into tOld
+      if tOld = pData then 
+         exit changeInfo -- they're the same, so don't bother changing anything
+      else if tOld is not empty then
+         -- negate node's old tag
+         put getTagKey(pMain,pBlerg) into tOldTag
+         put cr & "-" & tOldTag & "," & tRec+1 after pMain[pDoc][pRec]["step"]
+      end if
+      -- create turn
+      put tRec+2 & "," & tRec+1 into pMain[pDoc][tRec+1]["turn"]
+      put "changeInfo" into pMain[pDoc][tRec+1]["act"]
+      -- create new tag with new data
+      put pData into pMain[pDoc][tRec+2][pTag]
+      put tRec+1 & "," & tRec+1 into pMain[pDoc][tRec+2]["myTurn"]
+      put pRec & "," & tRec+1 into pMain[pDoc][tRec+2]["myStep"]
+      --      put tRec+2 into pMain[pDoc][0][lastKey]
+      -- point to new tag
+      put cr & tRec+2 & "," & tRec+1 after pMain[pDoc][pRec]["step"]
+   end if
+   put tRec+2 into pMain[pDoc][0][lastKey]
+end changeInfo
+
+command userFile
+   put "command userFile" into xX["write"];    devLog xX
+   -- ! at the moment this just stores whatever they picked and tries to display a preview
+   -- ! it should be easy to break
+   
+   put false into tShow
+   answer file "attach this file to" && stepString(sNowDoc,sNowItem)
```

Comment:
@matthewmaier You can use `answer file with type` so the user can only select the type you want. Also some of the extensions you are filtering out are platform specific. I'd stick with git, jpg and png.